chia/wallet/wallet_state_manager.py

```diff
@@ -1417,17 +1399,32 @@ async def _add_coin_states(
                             additions = [state.coin for state in children]
                             if len(children) > 0:
                                 fee = 0
-
+                                known_puzzle_hash: bool = await self.puzzle_store.puzzle_hash_exists(
+                                    coin_state.coin.puzzle_hash
+                                )
                                 to_puzzle_hash = None
+                                coin_spend: Optional[CoinSpend] = None
                                 # Find coin that doesn't belong to us
                                 amount = 0
-                                for coin in additions:
+                                for child_state in children:
+                                    coin: Coin = child_state.coin
                                     derivation_record = await self.puzzle_store.get_derivation_record_for_puzzle_hash(
                                         coin.puzzle_hash
                                     )
                                     if derivation_record is None:  # not change
                                         to_puzzle_hash = coin.puzzle_hash
                                         amount += coin.amount
+                                        if coin_spend is None:
+                                            coin_spend = await fetch_coin_spend_for_coin_state(coin_state, peer)
+
+                                        # Check if this is a Clawback coin
+                                        puzzle: Program = coin_spend.puzzle_reveal.to_program()
+                                        solution: Program = coin_spend.solution.to_program()
+                                        uncurried = uncurry_puzzle(puzzle)
+                                        clawback_metadata = match_clawback_puzzle(uncurried, puzzle, solution)
+                                        if clawback_metadata is not None and known_puzzle_hash:
+                                            # Add the Clawback coin as the interested coin for the sender
+                                            await self.add_interested_coin_ids([coin.name()])
```

Comment:
Can you explain why we need this here?

Reply:
As I mentioned in the comment above, if we want to recover the clawback for the sender, we need to manually add it to the interested_coin table since we don't own the puzzle hash and hint. The basic idea is we check the parent coin of the clawback coin. If it has a clawback coin owned by the current wallet, then we add it to the interested_coin so that we can receive the clawback coin state.

Reply:
Ideally, we want to hint both the sender and recipient puzzle hash. But we can only hint one puzzle hash and we are pointing to the recipient. That's why the recipient doesn't have this issue. As I mentioned above, unless we change how the hinting works, I don't think there is another way.